full-text-search-refactor/DocumentManagement/FilesInvertedIndexBuilder.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Text.RegularExpressions;
+
+namespace Mohaymen.FullTextSearch.DocumentManagement;
+
+public class FilesInvertedIndexBuilder : IInvertedIndexBuilder
+{
+    private readonly InvertedIndex _invertedIndex = new();
+
+    public FilesInvertedIndexBuilder IndexFilesWords(FileCollection fileCollection)
+    {
+        foreach (var filePath in fileCollection.GetFilesPath())
+        {
+            var keywords = ExtractKeywords(fileCollection.GetFileContent(filePath));
+            UpdateInvertedIndexMap(keywords, filePath);
+        }
+
+        return this;
+    }
+
+    public void UpdateInvertedIndexMap(List<Keyword> keywords, string filePath)
+    {
+        foreach (var keyword in keywords)
+        {
+            _invertedIndex.AddDocumentToKeyword(keyword, filePath);
+        }
+    }
+
+    public List<Keyword> ExtractKeywords(string fileContent)
+    {
+        return Regex.Split(fileContent, @"[^\w']+")
```

Comment:
move pattern to class private readonly property

Reply:
https://github.com/Star-Academy/Summer1403-SE-Team06/commit/68cceb97f7fa60862973ff1fa55602559acfea59